src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs

```diff
@@ -85,21 +97,154 @@ public static bool IsValidFormatArg(string formatString, CultureInfo formatCultu
                 return false;
             }
 
-            textFormatArgs.HasDateTimeFmt = textFormatArgs.FormatArg.IndexOfAny(new char[] { 'm', 'M', 'd', 'D', 'y', 'Y', 'h', 'H', 's', 'S', 'a', 'A', 'p', 'P' }) >= 0;
-            textFormatArgs.HasNumericFmt = textFormatArgs.FormatArg.IndexOfAny(new char[] { '0', '#' }) >= 0;
-            if (textFormatArgs.HasDateTimeFmt && textFormatArgs.HasNumericFmt)
+            bool hasNumericCharacters = false;
+            int decimalPointIndex = -1;
+            int sectionCount = 0;
+            var formatStr = textFormatArgs.FormatArg;
+            bool hasColonWithNum = false;
+
+            //Block "gen"/"general"
+            if (formatStr.IndexOf("gen", StringComparison.OrdinalIgnoreCase) >= 0)
             {
-                // Check if the date time format contains '0's after the seconds specifier, which
-                // is used for fractional seconds - in which case it is valid
-                var formatWithoutZeroSubseconds = _formatWithoutZeroSubsecondsRegex.Replace(textFormatArgs.FormatArg, m => m.Groups[1].Success ? string.Empty : m.Groups[1].Value);
-                textFormatArgs.HasNumericFmt = formatWithoutZeroSubseconds.IndexOfAny(new char[] { '0', '#' }) >= 0;
+                return false;
             }
 
-            if (textFormatArgs.HasDateTimeFmt && textFormatArgs.HasNumericFmt)
+            for (int i = 0; i < formatStr.Length; i++)
             {
-                return false;
+                // Do not allow format string start with '/'
+                if (i == 0 && formatStr[i] == '/')
```

Comment:
>i == 0 && formatStr[i] == '/' [](http://example.com/codeflow?start=20&length=29)

Can be moved outside the `for` so that the check isn't performed for every character in the format string

Reply:
yes, update